### PR TITLE
update canal with recent improvements to flannel

### PR DESCRIPTION
**Pull request by kwmonroe**
This aligns the flannel portion of canal with the recent fixes for [lp2004150](https://bugs.launchpad.net/charm-canal/+bug/2004150):

- align gh workflow and tox with the current charm-flannel impl
- use etcd3 calls to align with charm-flannel changes
- follow ck best practices for integration tests (overlay k8s-core with locally built charm/resources)
- use the word `align` as much as possible in this PR description

### CODE REVIEW

File: build-canal-resources.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ git clone $FLANNEL_REPO "${canal_temp}/flannel"
 pushd ${canal_temp}/flannel
 git checkout "$FLANNEL_COMMIT"
 ARCH="$ARCH" ./build-flannel-resources.sh
-mv flannel-*.gz ${canal_root}
+mv flannel-*.tar.gz ${canal_root}
```
Comment (addyess) [inline build-canal-resources.sh:31]: I suppose you'll need to update the FLANNEL_COMMIT to `d71d81e42c021a30323786d5cad44825efcf42c4`

File: .github/workflows/tox.yaml [hunk 2]
```diff
@@ -2,51 +2,108 @@ name: Run tests with Tox
 
 on:
   push:
-    branches: [main]
+    branches:
+      - main
```
Comment (addyess) [inline .github/workflows/tox.yaml:6]: it's kinda pointless to run these integration tests on merges to `main`

**Review by addyess — APPROVED** [review]
Thanks @kwmonroe these changes look perfect.  A really nice upgrade

File: .github/workflows/tox.yaml (hunk 2 above)
Comment (kwmonroe) [reply .github/workflows/tox.yaml]: you, alright!  i learned it by [watching you](https://github.com/charmed-kubernetes/charm-flannel/commit/1d18603292b88b659cc666ad0f876bb77c94f02b#diff-95557d53bf91069e59d82b9d8fcfaf52ec84a762858983edd5ef3c9b3e4c8191R4-R6).

nice catch, let's just test on PRs to main.

File: tox.ini [hunk 3]
```diff
@@ -15,42 +15,46 @@ deps =
...
     pytest-mock
-    charms.unit_test
     ipdb
+    git+https://github.com/juju-solutions/charms.unit_test/#egg=charms.unit_test
+# tox3 only passes through the upper-case versions by default, but some
+# programs, such as wget or pip, only honor the lower-case versions
+passenv =
+    http_proxy
+    https_proxy
+    no_proxy
```
Comment (kwmonroe) [inline tox.ini:25]: tox4 didn't like these on the same line, but newlines are [compatible with both tox3 and 4](https://github.com/tox-dev/tox/issues/2615).

**Pull request merged**